nixos/modules/services/monitoring/prometheus/exporters/dcgm.nix

```diff
@@ -0,0 +1,175 @@
+      '';
+    };
+    devices = mkOption {
+      type = types.str;
+      default = "f";
+      description = lib.mdDoc ''
+        Specify which devices dcgm-exporter monitors.
+
+        Possible values:
+
+        - `f` or
+        - `g[:id1[,-id2...]` or
+        - `i[:id1[,-id2...]`.
+
+        If an `id` list is used, then devices with match IDs must exist on the system. For example:
+
+        - `g` = Monitor all GPUs
+        - `i` = Monitor all GPU instances
+        - `f` = Monitor all GPUs if MIG is disabled, or all GPU instances if MIG is enabled.
+
+        Note: this rule will be applied to each GPU. If it has GPU instances, those will be monitored. If it doesn't, then the GPU will be monitored.
+
+        This is our recommended option for single or mixed MIG Strategies.
+
+        - `g:0,1` = monitor GPUs 0 and 1
+        - `i:0,2-4` = monitor GPU instances 0, 2, 3, and 4.
+
+        NOTE 1: `-i` cannot be specified unless MIG mode is enabled.
+
+        NOTE 2: Any time indices are specified, those indices must exist on the system.
+
+        NOTE 3: In MIG mode, only `-f` or `-i` with a range can be specified. GPUs are not assigned to pods and therefore reporting must occur at the GPU instance level.
+      '';
+    };
+    noHostname = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Omit the hostname information from the output, matching older versions.
+      '';
+    };
+    switchDevices = mkOption {
+      type = types.str;
+      default = "f";
+      description = lib.mdDoc ''
+        Specify which devices dcgm-exporter monitors.
+
+        Possible values:
+
+        - `f` or
+        - `g[:id1[,-id2...]` or
+        - `i[:id1[,-id2...]`.
+
+        If an `id` list is used, then devices with match IDs must exist on the system. For example:
+
+        - `g` = Monitor all GPUs
+        - `i` = Monitor all GPU instances
+        - `f` = Monitor all GPUs if MIG is disabled, or all GPU instances if MIG is enabled.
+
+        Note: this rule will be applied to each GPU. If it has GPU instances, those will be monitored. If it doesn't, then the GPU will be monitored.
+
+        This is our recommended option for single or mixed MIG Strategies.
+
+        - `g:0,1` = monitor GPUs 0 and 1
+        - `i:0,2-4` = monitor GPU instances 0, 2, 3, and 4.
+
+        NOTE 1: `-i` cannot be specified unless MIG mode is enabled.
+
+        NOTE 2: Any time indices are specified, those indices must exist on the system.
+
+        NOTE 3: In MIG mode, only `-f` or `-i` with a range can be specified. GPUs are not assigned to pods and therefore reporting must occur at the GPU instance level.
+      '';
+    };
+    fakeGpus = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Accept GPUs that are fake, for testing purposes only.
+      '';
+    };
+  };
+  serviceOpts.serviceConfig = {
+    AmbientCapabilities = ["CAP_SYS_ADMIN"];
```

Comment:
To me it seems like this service is only listening on a metrics endpoint of the `dcgm` service. Does it really need `CAP_SYS_ADMIN`? I could see the other one needing it though.